src/sentry/event_manager.py

```diff
@@ -753,26 +719,25 @@ def save(self, project_id, raw=False, assume_normalized=False):
                 group=group, environment=environment
             )
 
-        # save the event
-        try:
-            with transaction.atomic(using=router.db_for_write(Event)):
-                if options.get("store.save-event-skips-nodestore", True):
-                    event.data.save()
-                event.save()
-        except IntegrityError:
-            logger.info(
-                "duplicate.found",
-                exc_info=True,
-                extra={
-                    "event_uuid": event_id,
-                    "project_id": project.id,
-                    "group_id": group.id if group else None,
-                    "model": Event.__name__,
-                },
-            )
+        event.data.save()
 
-            if CHECK_DUPLICATES:
-                return event
+        if not options.get("store.skip-pg-save", False):
+            # save the event
+            try:
+                with transaction.atomic(using=router.db_for_write(Event)):
+                    event.data.save()
```

Comment:
This way you are calling event.data.save() twice as long as postgres save is not disabled. Is it intended ?

Reply:
No definitely not, been meaning to fix this. Thanks.